x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/filters/filters.tsx

```diff
@@ -0,0 +1,347 @@
+import { omit } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiDragDropContext,
+  EuiDraggable,
+  EuiDroppable,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiPanel,
+  euiDragDropReorder,
+  EuiButtonIcon,
+  EuiLink,
+  EuiText,
+  EuiButtonEmpty,
+  EuiForm,
+  EuiFormRow,
+  htmlIdGenerator,
+} from '@elastic/eui';
+import { updateColumnParam } from '../../../state_helpers';
+import { OperationDefinition } from '../index';
+import { FieldBasedIndexPatternColumn } from '../column_types';
+import { FilterPopover } from './filter_popover';
+import { IndexPattern } from '../../../types';
+import {
+  Query,
+  esKuery,
+  esQuery,
+  DataPublicPluginStart,
+} from '../../../../../../../../src/plugins/data/public';
+
+const generateId = htmlIdGenerator();
+
+// references types from src/plugins/data/common/search/aggs/buckets/filters.ts
+export interface Filter {
+  input: Query;
+  label: string;
+}
+export interface FilterValue {
+  input: Query;
+  label: string;
+  id: string;
+}
+
+const searchQueryLabel = i18n.translate('xpack.lens.indexPattern.searchQuery', {
+  defaultMessage: 'Search query',
+});
+
+export const defaultLabel = i18n.translate('xpack.lens.indexPattern.filters.label.placeholder', {
+  defaultMessage: 'All records',
+});
+
+// we don't have an access data plugin in places where we use this function,
+// what's why we have a fallback default input
+const getDefaultFilter = (data?: DataPublicPluginStart): Filter => ({
+  input: data
+    ? data.query.queryString.getDefaultQuery()
+    : {
+        query: '',
+        language: 'kuery',
+      },
+  label: '',
+});
+
+export const isQueryValid = (input: Query, indexPattern: IndexPattern) => {
+  try {
+    if (input.language === 'kuery') {
+      esKuery.toElasticsearchQuery(esKuery.fromKueryExpression(input.query), indexPattern);
+    } else {
+      esQuery.luceneStringToDsl(input.query);
+    }
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+interface DraggableLocation {
+  droppableId: string;
+  index: number;
+}
+
+export interface FiltersIndexPatternColumn extends FieldBasedIndexPatternColumn {
+  operationType: 'filters';
+  params: {
+    filters: Filter[];
+  };
+}
+
+export const filtersOperation: OperationDefinition<FiltersIndexPatternColumn> = {
+  type: 'filters',
+  displayName: searchQueryLabel,
+  priority: 3, // Higher than any metric
+  getPossibleOperationForField: ({ type }) => {
+    if (type === 'document') {
+      return {
+        dataType: 'string',
+        isBucketed: true,
+        scale: 'ordinal',
+      };
+    }
+  },
+  isTransferable: () => false,
+
+  onFieldChange: (oldColumn, indexPattern, field) => oldColumn,
+
+  buildColumn({ suggestedPriority, field, previousColumn }) {
+    let params = { filters: [getDefaultFilter()] };
+    if (previousColumn?.operationType === 'terms') {
+      params = {
+        filters: [
+          {
+            label: '',
```

Comment:
This should use `getDefaultQuery` as well, and then filling in query in the right format. Otherwise we might be picking the wrong query language.